arduino/readme.md

```diff
@@ -0,0 +1,37 @@
+# Arduino Lib notes
```

Comment:
Can we add some commentary here (or perhaps another appropriate readme) that indicate the purpose of the arduino tree and that it is not part of a native (non Arduino tools) build?

Reply:
Yes, this is currently a dumping ground for notes as I'm working on porting this over. It is really ugly still..